ui/uihandler.go

```diff
@@ -275,29 +278,36 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 			log.LogVf("adding header %v", header)
 			if firstHeader {
 				// If there is at least 1 non empty H passed, reset the header list
-				opts.ResetHeaders()
+				httpopts.ResetHeaders()
 				firstHeader = false
 			}
-			err := opts.AddAndValidateExtraHeader(header)
+			err := httpopts.AddAndValidateExtraHeader(header)
 			if err != nil {
 				log.Errf("Error adding custom headers: %v", err)
 			}
 		}
-		onBehalfOf(opts, r)
-		o := fhttp.HTTPRunnerOptions{
-			RunnerOptions:      ro,
-			HTTPOptions:        *opts,
-			AllowInitialErrors: true,
-		}
+		onBehalfOf(httpopts, r)
 		if !JSONOnly {
 			flusher.Flush()
 		}
-		res, err := fhttp.RunHTTPTest(&o)
-		uiRunMapMutex.Lock()
-		delete(runs, runid)
-		uiRunMapMutex.Unlock()
+		var res periodic.HasRunnerResult
+		var err error
+		if runner == modegrpc {
+			o := fgrpc.GRPCRunnerOptions{
+				RunnerOptions: ro,
+				Destination:   url,
+			}
+			res, err = fgrpc.RunGRPCTest(&o)
+		} else {
+			o := fhttp.HTTPRunnerOptions{
+				HTTPOptions:        *httpopts,
+				RunnerOptions:      ro,
+				AllowInitialErrors: true,
+			}
+			res, err = fhttp.RunHTTPTest(&o)
+		}
 		if err != nil {
-			log.Errf("Init error %+v : %v", o, err)
+			log.Errf("Init error %+v : %v", ro, err)
```

Comment:
we do lose some visibility here. but I did not think this merited creation of a `HasRunnerOptions` interface

Reply:
it's ok, though maybe add the input url and runner var